### PR TITLE
Update RStudio publisher to fix integration

### LINKED ISSUE #15810: GitHub Desktop not finding RStudio as External Editor
### The problem

I recently upgraded to RStudio 2022.12.0 Build 353 and Github Desktop can no longer find RStudio as an external editor. I could previously open my RStudio project directly from Github Desktop, but now I have to open the folder and just open the .rproj 

This is a bit inconvenient. I try to go to the options to set an external editor and RStudio is not even available.

Anyone else having this issue?

### Release version

2022.12.0 Build 353

### Operating system

Windows 11

### Steps to reproduce the behavior

_No response_

### Log files

_No response_

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by steveward**
Closes #15810 

## Description

- Update publisher from 'RStudio' to 'Posit Software' to fix the RStudio integration
- Tested this in a dev build on Windows and confirmed it fixed the integration

## Release notes
No notes

### CODE REVIEW

**Review by Mar181-arc — COMMENTED** [review]
Leave a rewiev

**Review by tidy-dev — DISMISSED** [review]

**Comment by sergiou87** [comment]
Won't this change break compatibility with older versions? Maybe we need a new whole entry that duplicates everything and only changes the publisher (either that or allowing multiple publishers per entry 🤔 )

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -357,7 +357,7 @@ const editors: WindowsExternalEditor[] = [
     registryKeys: [Wow64LocalMachineUninstallKey('RStudio')],
     installLocationRegistryKey: 'DisplayIcon',
     displayNamePrefix: 'RStudio',
-    publisher: 'RStudio',
+    publisher: 'Posit Software',
```
Comment (sergiou87) [inline app/src/lib/editors/win32.ts]: After @voidei 's work in #15585 we can just have multiple publishers, which should solve this problem! 🎉  😄 

```suggestion
    publishers: ['RStudio', 'Posit Software'],
```

**Review by sergiou87 — CHANGES_REQUESTED** [review]

**Comment by steveward** [comment]
Thanks @sergiou87, I should've thought of that! 😅

**Comment by Mortezaegoe** [comment]
> ssaaa[]() 

> ``

**Comment by Mortezaegoe** [comment]
R

**Review by sergiou87 — APPROVED** [review]
Thanks for that change! It works like a charm with both new and old versions 🙌

**Comment by ars17psu** [comment]
I have R Studio 2022.12.0+353 and R 4.2.2 GUI 1.79 High Sierra build running on a Mac with MacOS Ventura. An option to open RStudio or Posit Software is not available in GitHub Desktop Version 3.1.6 (x64) or Version 3.1.7 (x64). Any assistance you can provide would be much appreciated.

**Pull request merged**